sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -2552,7 +2512,19 @@ internal virtual void SetNameFieldsIfNull()
                                 filesSuccessfulCount += currentFilesSuccessfulCount;
                                 failureCount += currentFailureCount;
 
-                                if (progressHandler != null)
+                                if ((currentFailureCount > 0) &&
+                                    ((options?.ContinueOnFailure.GetValueOrDefault(false) == false) ||
+                                    (batchFailures == default)))
+                                {
```

Comment:
I think we can simplify this condition to currentFailureCount && batchFailures ==default.

Also plz add tests where we have failures that span across many batches - let's assert which batch is taken there.